### ISSUE TITLE
Tabs - State and A11y errors

### ISSUE THREAD

**Issue by jgonzmoya** [Capgemini/dcx-react-library#705]
**Describe the bug**

1. A11y Error: The hardcoded tab-index value at the tabcontent level is causing accessibility issues because it is automatically taking focus when tabbing through the content. Focus delegation should occur at the child level. For example, when there is a list of buttons as children, the focus currently lands on the tabcontent itself on the first tab press, instead of on the first focusable element, which should be the first button.

2. State Error: The activeKey is not reactive to changes; it always takes the first value on component mount and only responds to clicks. You cannot change the activeKey from outside the component and have it activate the corresponding tab passed via props. The internal state causes the component to take a value only during initialization, and it can only be changed either by forcing a component unmount or through a click event. This behavior is problematic because, in order to achieve the desired functionality, we are forced to unmount the component, which results in losing the scroll position of the tabs, for example.

**To Reproduce**
A11y Error:
1. Go to Tabs component
2. Click on tab keyboard
3. you can see that the content is focus

State Error:
1. Go to Tabs component
2. Change the activeKey in code
3. you can see the selected tab not change

**Expected behavior**
A11y Error: The tab content should not have a hardcoded tabindex; instead, it should allow its children to control focus behavior.

State Error:We should be able to change the activeKey reactively without having to force a component unmount.

**Screenshots**
A11y Error:

-Code screenshot:

![Image](https://github.com/user-attachments/assets/bcab68d5-41a2-4323-b662-16829596f150)

-Issue screenshot:

![Image](https://github.com/user-attachments/assets/7aff5bbf-5fd7-49bb-87e7-a132316b7cb4)

![Image](https://github.com/user-attachments/assets/941795e3-c981-4f78-b594-d0ef6158bfd9)

State Error:

![Image](https://github.com/user-attachments/assets/080d9f8a-1e1c-4501-9183-a070253b5042)

**Desktop (please complete the following information):**
 - OS: All
 - Browser All
 - Version All

**Smartphone (please complete the following information):**
 - Device: All
 - OS: All
 - Browser All
 - Version All

**Additional context**

```bash
git checkout release/1.1.0
git pull
git checkout -b 'fix/tabs-state-a11y'
```
